src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -169,38 +170,105 @@ def prepare_or_run_pipeline(
-                **kwargs,
+            # Retrieve Processor arguments provided in the Step settings.
+            processor_args_for_step = step_settings.processor_args or {}
+
+            # Set default values from configured orchestrator Component to arguments
+            # to be used when they are not present in processor_args.
+            processor_args_for_step.setdefault(
+                "instance_type", step_settings.instance_type
+            )
+            processor_args_for_step.setdefault(
+                "role",
+                step_settings.processor_role or self.config.execution_role,
+            )
+            processor_args_for_step.setdefault(
+                "volume_size_in_gb", step_settings.volume_size_in_gb
+            )
+            processor_args_for_step.setdefault(
+                "max_runtime_in_seconds", step_settings.max_runtime_in_seconds
+            )
+            processor_args_for_step.setdefault(
+                "tags",
+                [step_settings.processor_tags]
+                if step_settings.processor_tags
+                else None,
             )
 
+            # Set values that cannot be overwritten
+            processor_args_for_step["image_uri"] = image
+            processor_args_for_step["instance_count"] = 1
+            processor_args_for_step["sagemaker_session"] = session
+            processor_args_for_step["entrypoint"] = entrypoint
+            processor_args_for_step["base_job_name"] = orchestrator_run_name
+            processor_args_for_step["env"] = environment
+
+            # Construct S3 inputs to container for step
+            inputs = None
+
+            if step_settings.input_data_s3_uri is None:
+                pass
+            elif isinstance(step_settings.input_data_s3_uri, str):
+                inputs = [
+                    ProcessingInput(
+                        source=step_settings.input_data_s3_uri,
+                        destination="/opt/ml/processing/input",
+                        s3_input_mode=step_settings.input_data_s3_mode,
+                    )
+                ]
+            elif isinstance(step_settings.input_data_s3_uri, dict):
+                inputs = []
+                for channel, s3_uri in step_settings.input_data_s3_uri.items():
+                    inputs.append(
+                        ProcessingInput(
+                            source=s3_uri,
+                            destination=f"/opt/ml/processing/input/{channel}",
+                            s3_input_mode=step_settings.input_data_s3_mode,
+                        )
+                    )
+
+            # Construct S3 outputs from container for step
+            outputs = None
+
+            if step_settings.output_data_s3_uri is None:
+                pass
+            elif isinstance(step_settings.output_data_s3_uri, str):
+                outputs = [
+                    ProcessingOutput(
+                        source="/opt/ml/processing/output",
```

Comment:
Is this something we want configurable as well?

Reply:
We could do that, but then we would deviate a bit from what SageMaker discusses here: https://docs.aws.amazon.com/sagemaker/latest/dg/model-access-training-data.html - in the image, the location in the filesystem specifies `/opt/ml/input/data/<channel>` (I assumed the same for output).
To be highly precise, I would probably need to add `./data` to all `ProcessingInput` and `ProcessingOutput` references (let me do that right now), but I would suggest to keep the hardcoded ref as-is? What do you think?